dotnet/src/Connectors/Connectors.Memory.Qdrant/QdrantVectorRecordStore.cs

```diff
@@ -0,0 +1,382 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Grpc.Core;
+using Microsoft.SemanticKernel.Memory;
+using Qdrant.Client;
+using Qdrant.Client.Grpc;
+
+namespace Microsoft.SemanticKernel.Connectors.Qdrant;
+
+/// <summary>
+/// Service for storing and retrieving vector records, that uses Qdrant as the underlying storage.
+/// </summary>
+/// <typeparam name="TRecord">The data model to use for adding, updating and retrieving data from storage.</typeparam>
+public sealed class QdrantVectorRecordStore<TRecord> : IVectorRecordStore<ulong, TRecord>, IVectorRecordStore<Guid, TRecord>
+    where TRecord : class
```

Comment:
This needs unit tests also.

Reply:
Same as above, will do after the unit test project refactor.